src/math.rs

```diff
@@ -0,0 +1,24 @@
+/// Divide `n` by `divisor`, and round up to the nearest integer
+/// if not evenly divisable.
+#[inline]
+pub fn div_round_up(n: usize, divisor: usize) -> usize {
+    if n == 0 {
+        0
```

Comment:
It's a _little_ bit sketchy to have `div_round_up(0, 0) -> 0`, but the places we're using it will never have a 0 `divisor`.  Maybe just add a `debug_assert!(divisor != 0)`?